docs/contributing/github-ci.md

```diff
@@ -0,0 +1,31 @@
+# GitHub CI
+
+> This guide explains how to publish a forked version of Directus on NPM, dockerhub and GHCR. You need to first
```

Comment:
These are more internal process docs than user facing docs.

Using a fork is not something we recommend doing, and is out of scope for what we want to officially support.

Reply:
May I ask why? For most other projects forking is straightforward and doesn't break anything. Here it causes undefined behavior, because you **can** fork and you **can** build an image from a fork but that image will not contain the changes you made.

If you're just worried about getting issues here that are actually only occuring because of some unofficial code changes then a disclaimer would be enough (and a checkbox in the issue template "i am using the official directus build")

Reply:
> If you're just worried about getting issues here that are actually only occuring because of some unofficial code changes then a disclaimer would be enough (and a checkbox in the issue template "i am using the official directus build")

Yup! By "officially supporting" a fork, we effectively put ourselves into a corner where every change in the internal codebase is a potential breaking change for the users who've forked. Something as simple as changing a class on an element could break someones fork.

If we add a massive disclaimer at the top that this effectively "voids the warranty" we can go ahead I suppose, but even then I'd rather have users solve problems through extensions and bug fixes in the repo rather than going ahead with custom forks, as that inevitably causes fragmentation in the ecosystem and additional support/maintenance overhead for the main project

Reply:
> but even then I'd rather have users solve problems through extensions and bug fixes in the repo rather than going ahead with custom forks, as that inevitably causes fragmentation in the ecosystem and additional support/maintenance overhead for the main project

100% my opinion (and i'm having a hell of a time with keeping https://github.com/directus/directus/pull/3592 up to date :P), but there are just things that extensions can't do. This should be part of the disclaimer aswell though... I'll write something up in the documentation (ETA a few hours) that you can check if it meets the requirements.

Reply:
Yup! That's very true, but the couple of things that are out of scope for extensions should be out of 80/20 as well, so would be out of scope for our efforts